src/main/java/Hailey.java

```diff
@@ -0,0 +1,73 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Hailey {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        List<Task> tasks = new ArrayList<>();
+
+        System.out.println("Hello! I'm Hailey");
+        System.out.println("What can I do for you?");
+
+        while (true) {
+            String input = scanner.nextLine();
+
+            if ("bye".equalsIgnoreCase(input)) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            } else if ("list".equalsIgnoreCase(input)) {
+                if (tasks.isEmpty()) {
+                    System.out.println("No tasks added yet.");
+                } else {
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < tasks.size(); i++) {
+                        Task task = tasks.get(i);
+                        System.out.printf(" %d.%s %s%n", i + 1, task.getStatusIcon(), task.getDescription());
+                    }
+                }
+            } else if (input.startsWith("mark")) {
+                String[] parts = input.split(" ");
+                if (parts.length == 2 && isNumeric(parts[1])) {
+                    int index = Integer.parseInt(parts[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        Task task = tasks.get(index);
+                        task.markAsDone();
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.printf("  %s %s%n", task.getStatusIcon(), task.getDescription());
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else {
+                    System.out.println("Invalid command format.");
+                }
+            } else if (input.startsWith("unmark")) {
+                String[] parts = input.split(" ");
+                if (parts.length == 2 && isNumeric(parts[1])) {
+                    int index = Integer.parseInt(parts[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        Task task = tasks.get(index);
+                        task.markAsUndone();
+                        System.out.println("OK, I've marked this task as not done yet:");
+                        System.out.printf("  %s %s%n", task.getStatusIcon(), task.getDescription());
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else {
+                    System.out.println("Invalid command format.");
+                }
+            } else {
+                tasks.add(new Task(input));
+                System.out.println("added: " + input);
+            }
+        }
+
+        scanner.close();
+    }
```

Comment:
This code block is deeply nested, and needs to be worked on.

In general, this method is too long, and it is preferred for methods to be shorter than 30 lines of code.

Please work on abstracting this chunk of code.

References:
1. https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting
2. https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods